### PR TITLE
Compressor plugin hideable controls

### LINKED ISSUE #6830: Native compressor plugin display enhancement
### Enhancement Summary

Be able to see the graph of the compression effect without the effect controls widget masking a big part of it.

#### Justification

I like graphs.

I’m not sure if the realtime graph of this plugin has been conceived as just a decoration, or if I’m to dumb to have been able to find how to do it, but it seems there is no possibility to hide the controls to get a full view of the graph.

Also, the possibility to hide the graph to show only the effect controls would also be welcome, in a CPU cycle saving goal. But that’s another thing.

### PR DESCRIPTION
**Pull request by szeli1**
Closes #6830

With this change, if you double click in the compressor plugin/effect then the controls will be hidden similarly to the Equalizer.

### CODE REVIEW

File: plugins/Compressor/CompressorControlDialog.h [hunk 1]
```diff
@@ -214,6 +220,8 @@ private slots:
 	PixmapButton * feedbackButton;
 	PixmapButton * lookaheadButton;
 
+	bool guiVisibility;
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.h:223]:
```suggestion
	bool m_guiVisibility = true;
```
Coding convention for data members.

Also moved the initialization to here instead of in the constructor.

File: plugins/Compressor/CompressorControlDialog.cpp [hunk 2]
```diff
@@ -310,6 +310,8 @@ CompressorControlDialog::CompressorControlDialog(CompressorControls* controls) :
 	connect(&m_controls->m_lookaheadModel, SIGNAL(dataChanged()), this, SLOT(lookaheadChanged()));
 	connect(&m_controls->m_limiterModel, SIGNAL(dataChanged()), this, SLOT(limiterChanged()));
 
+	guiVisibility = true;
+
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.cpp:314]:
```suggestion
```
Can move initialization to header

File: plugins/Compressor/CompressorControlDialog.cpp [hunk 3]
```diff
@@ -690,6 +692,59 @@ void CompressorControlDialog::drawGraph()
 }
 
 
+void CompressorControlDialog::mouseDoubleClickEvent(QMouseEvent * event)
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.cpp:695]:
```suggestion
void CompressorControlDialog::mouseDoubleClickEvent(QMouseEvent* event)
```
style

File: plugins/Compressor/CompressorControlDialog.cpp [hunk 4]
```diff
@@ -690,6 +692,59 @@ void CompressorControlDialog::drawGraph()
...
+{
+	setAllGuiVisible(!getGuiVisibility());
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.cpp:697]:
```suggestion
	setGuiVisibility(!m_guiVisibility);
```

File: plugins/Compressor/CompressorControlDialog.cpp [hunk 5]
```diff
@@ -690,6 +692,59 @@ void CompressorControlDialog::drawGraph()
...
+}
+
+
+void CompressorControlDialog::setAllGuiVisible(bool isVisible)
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.cpp:701]:
```suggestion
void CompressorControlDialog::setGuiVisibility(bool isVisible)
```
I prefer this name

File: plugins/Compressor/CompressorControlDialog.cpp [hunk 6]
```diff
@@ -690,6 +692,59 @@ void CompressorControlDialog::drawGraph()
...
+	unlinkedButton->setVisible(isVisible);
+	maximumButton->setVisible(isVisible);
+	averageButton->setVisible(isVisible);
+	minimumButton->setVisible(isVisible);
+	blendButton->setVisible(isVisible);
+	autoMakeupButton->setVisible(isVisible);
+	auditionButton->setVisible(isVisible);
+	feedbackButton->setVisible(isVisible);
+	lookaheadButton->setVisible(isVisible);
+	guiVisibility = isVisible;
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.cpp:744]:
```suggestion
	m_guiVisibility = isVisible;
```
To match coding conventions

File: plugins/Compressor/CompressorControlDialog.h [hunk 7]
```diff
@@ -108,6 +108,12 @@ private slots:
 	void drawKneePixmap2();
 	void drawMiscPixmap();
 	void drawGraph();
+	void mouseDoubleClickEvent(QMouseEvent *event) override;
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.h:111]:
```suggestion
	void mouseDoubleClickEvent(QMouseEvent* event) override;
```
style

File: plugins/Compressor/CompressorControlDialog.h [hunk 8]
```diff
@@ -108,6 +108,12 @@ private slots:
...
+	inline bool getGuiVisibility()
+	{
+		return guiVisibility;
+	}
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.h:115]:
```suggestion
```
I would just remove this method since it's only used once and from within this class.

File: plugins/Compressor/CompressorControlDialog.h [hunk 9]
```diff
@@ -108,6 +108,12 @@ private slots:
...
+	void setAllGuiVisible(bool isVisible);
```
Comment (messmerd) [inline plugins/Compressor/CompressorControlDialog.h:116]:
```suggestion
	void setGuiVisibility(bool isVisible);
```

**Review by messmerd — COMMENTED** [review]
Haven't tested it yet - this is a style review

File: plugins/Compressor/CompressorControlDialog.cpp [hunk 10]
```diff
@@ -690,6 +690,79 @@ void CompressorControlDialog::drawGraph()
...
+
+		m_blendKnob->setVisible(m_controls->m_stereoLinkModel.value() == 4);
+		m_blendEnabledLabel->setVisible(m_controls->m_stereoLinkModel.value() == 4);
+
+		m_lookaheadLengthKnob->setVisible(m_controls->m_lookaheadModel.value());
+		m_lookaheadEnabledLabel->setVisible(m_controls->m_lookaheadModel.value());
+
+		m_ratioKnob->setVisible(!m_controls->m_limiterModel.value());
+		m_ratioEnabledLabel->setVisible(!m_controls->m_limiterModel.value());
+	}
```
Comment (sakertooth) [inline plugins/Compressor/CompressorControlDialog.cpp:728]: Why is this needed? Can't we just simply set all the widgets visibility to `isVisible`?

**Review by sakertooth — COMMENTED** [review]
Tested and working, quick question though.

File: plugins/Compressor/CompressorControlDialog.cpp (hunk 10 above)
Comment (szeli1) [reply plugins/Compressor/CompressorControlDialog.cpp:728]: Some controls are hidden by default such as the rms Knob. It is only visible when the rms button is active.

**Review by sakertooth — APPROVED** [review]

**Comment by sakertooth** [comment]
@messmerd do you still want to test this?
